http/query_service.go

```diff
@@ -1,242 +0,0 @@
-package http
-
-import (
-	"bytes"
-	"context"
-	"encoding/json"
-	"net/http"
-
-	"github.com/influxdata/flux"
-	"github.com/influxdata/flux/csv"
-	icontext "github.com/influxdata/influxdb/context"
-	"github.com/influxdata/influxdb/query"
-	"github.com/julienschmidt/httprouter"
-	"github.com/prometheus/client_golang/prometheus"
-	"go.uber.org/zap"
-)
-
-const (
-	queryPath = "/api/v2/querysvc"
-
-	queryStatisticsTrailer = "Influx-Query-Statistics"
-)
-
-// QueryHandler is a low-level query service handler.
-type QueryHandler struct {
-	*httprouter.Router
-
-	Logger *zap.Logger
-
-	csvDialect csv.Dialect
-
-	QueryService     query.QueryService
-	CompilerMappings flux.CompilerMappings
-}
-
-// NewQueryHandler returns a new instance of QueryHandler.
-func NewQueryHandler() *QueryHandler {
```

Comment:
This makes this into a pretty big breaking change in my opinion and we should caution against doing this. It's something we're going to need to discuss if we accept this change.

I think the easiest thing to do is to keep this code and have `QueryHandler` change to using proxy query service. That's a breaking change for Go code, but the HTTP API remains the same and it has a far smaller surface area.

Reply:
I think a goal that @nathanielc had with this refactoring was to make it so that the code that gateway uses to talk to `queryd` and `transpilerde` is the same to reduce technical debt.  

We should discuss with Nathaniel when we're all around.

Reply:
I think that's fine and a good goal. My only issue is that it's combining too many things in this PR which makes it more complex to review and adds additional points where it may break.

We can discuss this with Nathaniel though to see if it's worth going down that road.

Reply:
This change was quite big and effected lots of different bits,as such I saw it as valuable to delete this dead code as that was the intent of this change. By deleting the code we are confident that we put everything back together correctly without relying on this old code.